kratos/solving_strategies/schemes/residual_based_bdf_displacement_scheme.h

```diff
@@ -145,6 +147,70 @@ class ResidualBasedBDFDisplacementScheme
     ///@name Operations
     ///@{
 
+    /**
+     * @brief It initializes time step solution. Only for reasons if the time step solution is restarted
+     * @param rModelPart The model part of the problem to solve
+     * @param rA LHS matrix
+     * @param rDx Incremental update of primary variables
+     * @param rb RHS Vector
+     */
+    void InitializeSolutionStep(
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY;
+
+        // The current process info
+        const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+        BDFBaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
+
+        // Updating time derivatives (nodally for efficiency)
+        const int num_nodes = static_cast<int>( rModelPart.Nodes().size() );
+        const auto it_node_begin = rModelPart.Nodes().begin();
+
+        // Getting dimension
+        const std::size_t dimension = r_current_process_info.Has(DOMAIN_SIZE) ? r_current_process_info.GetValue(DOMAIN_SIZE) : 3;
```

Comment:
TBD

Reply:
TDB?

Reply:
To Be Done (read ... to be changed for automatic detection as we commented elsewhere)

Reply:
Well can be detected with the geometries, but you can have a problem with just nodes, so is not the proper way.

Reply:
The constructor doesn't know the model part, then the domain_size is not known...